dbt/include/global_project/macros/materializations/snapshots/snapshot.sql

```diff
@@ -39,6 +39,22 @@
 
       {{ adapter.valid_snapshot_target(target_relation, columns) }}
 
+      {# Raise an exception if the user has selected the new_record mode for
```

Comment:
Wouldn't we have to add new parameters (i.e. the snapshot strategy or a flag indicating whether or not to expect 'dbt_is_deleted) to the function to accomplish that? Is that safe to change? It's been a couple weeks since I implemented this, but I think the function signature is why I put the logic here.

Reply:
We decided to add a new base adapter method valid_snapshot_target_with_strategy() and use it for this case.